src/Processors/QueryPlan/CreatingSetsStep.h

```diff
@@ -51,7 +51,7 @@ class CreatingSetsStep : public IQueryPlanStep
 
 void addCreatingSetsStep(
     QueryPlan & query_plan,
-    SubqueriesForSets subqueries_for_sets,
+    PreparedSets & prepared_sets,
```

Comment:
Changed to `PreparedSetsPtr prepared_sets` because we check `(!prepared_sets || prepared_sets->empty())` inside `addCreatingSetsStep`. Also renamed `moveSubqueries` to `detachSubqueries`. We do not invalidate the whole object, but just get plans for sub-queries, because we need to use it just once (added it to comments):

https://github.com/ClickHouse/ClickHouse/blob/8eecb9ef8236d6b87f67628b5fd21f1f573320cc/src/Interpreters/PreparedSets.h#L81-L84